Jenkinsfile

```diff
@@ -74,9 +74,82 @@ Sleep-seconds: 2'''
+                    !skipStage()
+                }
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'TEST_BRANCH'
+                        values 'master',
+                               'release/2.6'
+                    }
+                }
+                when {
+                    beforeAgent true
+                    expression {
+                        // Need to pass the stage name: https://issues.jenkins.io/browse/JENKINS-69394
+                        !skipStage([stage_name: 'Test Library',
+                                    axes: env.TEST_BRANCH.replaceAll('/', '-')])
+                    }
+                }
+                stages {
+                    stage('Test Library') {
+                        steps {
+                            // Normally we do not want to call pipeline-lib methods
+                            // from trusted-pipeline-lib, but this is needed
+                            // for properly running the DAOS testing.
+                            setupDownstreamTesting('daos-stack/daos', env.TEST_BRANCH,
+                                                   (cachedCommitPragma('Test-skip-build', 'false') == 'true' ?
+                                                        'Skip-build: true' : '') +
+                                                   (cachedCommitPragma('Skip-downstream-test', 'false') == 'true' ?
+                                                            '\nSkip-test: true' : ''))
+                            build job: 'daos-stack/daos/' + setupDownstreamTesting.test_branch(env.TEST_BRANCH),
+                                  parameters: [string(name: 'TestTag',
+                                                      value: 'load_mpi test_core_files'),
+                                               // Maybe should only do this if Test-tag: provisioning?
+                                               string(name: 'CI_RPM_TEST_VERSION',
+                                                      value: daosLatestVersion(env.TEST_BRANCH)),
+                                               string(name: 'BuildPriority', value: '2'),
+                                               booleanParam(name: 'CI_FI_el8_TEST', value: true),
+                                               booleanParam(name: 'CI_MORE_FUNCTIONAL_PR_TESTS', value: true),
+                                               booleanParam(name: 'CI_FUNCTIONAL_el9_TEST', value: true),
+                                               booleanParam(name: 'CI_FUNCTIONAL_el8_TEST', value: true),
+                                               booleanParam(name: 'CI_FUNCTIONAL_leap15_TEST', value: true),
+                                               booleanParam(name: 'CI_medium_TEST', value: false),
+                                               booleanParam(name: 'CI_large_TEST', value: false)
+                                              ]
```

Comment:
Wondering why the differences in the parameters specified here vs. https://github.com/daos-stack/pipeline-lib/blob/29f5c4fb82bae3164a979970b9d2cb922f9e8746/Jenkinsfile#L758-L777.

Indeed, rather than duplicating code here, I wonder if this should be wrapped into a common function.

Reply:
The ones in pipeline-lib are not up to date and need to be also fixed.   If you specify a parameter to the build_job that  does not exist, Jenkins writes log message for each one in the jenkins.log file.

As much as I don't like calling pipelilne-lib routines from trusted-pipeline-lib and system-pipeline-lib, it seems to make sense to put a routine in pipeline-lib to set these parameters.